### DISCUSSION TITLE
Help modifying the positions of the items.

### DISCUSSION THREAD

**Discussion by LupuC** [prisma/prisma#23820] [Q&A] [answered]
### Question

Hi, so, basically, I have 2 models.

1. Section with:
- id mongodb id
- name string
- position int
- lists List[] @relation("TaskToSection")

2. List
- id mongodb id
- name string
- position int
- section   Section  @relation(name: "ListToSection", fields: [sectionId], references: [id]) // Relationship to the Section model
- sectionId String   @db.ObjectId // Foreign key to associate the task with a section

I want to reorder the list within a section, and I don't know how to do that.
On my nextjs api endpoint, i have this data:
list, sectionId, oldPosition, newPosition.

The problem is that, I couldn't find a solution to edit all the others lists positions.

I've tried to get (based on the sectionId) the section, then all the tasks, and loop through them to update the position.. didn't work, and I don't know if it's the best idea...

```js
export async function PUT(request: NextRequest) {
    try {
        const { list, sectionId, oldPosition, newPosition } = await request.json();

        console.log(
            "Moved list: ",
            list.id,
            "from: ",
            oldPosition,
            "to: ",
            newPosition,
            "in section: ",
            sectionId
          );

        
          

        return NextResponse.json({ message: "List updated successfully" });

    }catch (error) {
        return NextResponse.json({ error: 'Internal server error' }, { status: 500 });
    }
}
```

**Comment by ludralph** [comment]
Hi @LupuC 👋 

Thank you for raising this question.

Could you please provide more details on the specific behavior you're expecting when reordering the lists within a section? Can you also describe the specific problem you encountered when looping through all the tasks to update positions? Were there any error messages or unexpected behavior? Lastly, is this the correct schema you are using below?
```prisma
model Section {
  id String @id @default(auto()) @map("_id") @db.ObjectId
  name String
  position Int
  lists List[] @relation("ListToSection")
}

model List{
  id String @id @default(auto()) @map("_id") @db.ObjectId
  name String
  position Int
  section Section @relation("ListToSection", fields: [sectionId], references: [id])
  sectionId String  @db.ObjectId
}
```

**Comment by LupuC** [comment]
Yes, that is the schema. I'm using React-Beautiful-Dnd, to reorder the sections and lists (or move lists within sections), but, I cannot understand what would be the logic for that.

Now, take a look at my section implementation, that only changes the positions:

```
export async function PUT(request: NextRequest) {
    try {
        const { sections } = await request.json();

        await Promise.all(sections.map(async (section: SectionModel) => {
            await prisma.section.update({ where: { id: section.id }, data: { position: section.position } });
        }));

        return NextResponse.json({ message: "Sections updated successfully" });

    }catch (error) {
        return NextResponse.json({ error: 'Internal server error' }, { status: 500 });
    }
}
```

That's pretty simple. I can do that with lists too, if I reorder them, but, I can't find a login when I move lists between sections..
I mean, as sections, reordering something, it's pretty simple. Find out all the items with their positions, and update them all. But, when moving from a section to another, I have to:
Update all lists from previous sections, then update the positions of list from the next section.

Example:
Section 1:               Section 2:
List1(pos0)             ListA(pos0)
List2(pos1)             ListB(pos1)
List3(pos2)             ListC(pos2)
List4(pos3)             ListD(pos3)
List5(pos4)             ListE(pos4)

If I move the List3 (pos2), to section 2, on position 3, then, i have to decrease List4 and List5 position with 1, then, increase the ListD and ListE position by 1, while updating the new sectionId for that list..

I know it's pretty overwhelming, and if you need, I can show you the full code.

**Comment by LupuC** [comment, marked as answer]
Nevermind, I rethinked the logic of what I wanted to get, and I came with this solution, for now:

```
export async function PUT(request: NextRequest) {
    try {
        const { data } = await request.json();
        console.log(data)
        await Promise.all(data.map(async (section: SectionModel) => {
            await prisma.section.update({ where: { id: section.id }, data: { position: section.position } });
            await Promise.all(section.lists.map(async (list: any) => {
                await prisma.list.update({ where: { id: list.id }, data: { position: list.position, sectionId: section.id } });
            }));
        }));
        return NextResponse.json({ message: "Data updated successfully" });

    }catch (error) {
        return NextResponse.json({ error: 'Internal server error' }, { status: 500 });
    }
}
```
